tests/fp-string-error.test.ts

```diff
@@ -0,0 +1,22 @@
+import { describe, expect, it } from "bun:test"
+import { getTestFixture } from "./fixtures" // Adjust path based on your structure
+
+describe("FP String Error Tests", () => {
+  it("should throw an error for invalid footprint function", async () => {
+    const { fp } = await getTestFixture("slop1")
+
+    const invalidFootprint = "invalidFunction"
+
+    try {
+      fp.string(invalidFootprint).soup()
+      expect.fail("Expected an exrror to be thrown, but no error was thrown")
+    } catch (error) {
+      expect(error).toBeInstanceOf(Error)
+
+      expect(error.message).toContain("Function not found for footprinter")
```

Comment:
You have to test that “invalidFunction” appears in the error output, that is the key part of the implementation that you added, so you should test it

Reply:
Better to just test an exact match for the whole message this test is hard to read. Just tell ne as the reviewer the full message with a toEqual assertion